react/src/containers/EditMovieModal.js

```diff
@@ -0,0 +1,126 @@
+import React, {useEffect, useState} from 'react';
+import {Modal, Button, Form, Col} from 'react-bootstrap';
+import {useFormik} from 'formik';
+import {useSelector, useDispatch} from 'react-redux';
+import Loader from '../component/Loader';
+import {validationSchema, initialValues} from '../config/movieValidation';
+import { genreSelector, movieSelector } from '../store/selectors/MovieSelector';
+import { editMovie } from '../store/actions/MovieActions';
+import { userSelector } from '../store/selectors/UserSelector';
+
+
+export default function EditMovieModal() {
+    const [show, setShow] = useState(false);
+    const movie = useSelector(movieSelector());
+
+    const genres = useSelector(genreSelector());
+    
+    const dispatch = useDispatch();
+  
+    const handleClose = () => setShow(false);
+    const handleShow = () => setShow(true);
+    
+    const formik = useFormik({
+        initialValues: initialValues({id: movie.id, title: movie.title, description: movie.description,
+             image: movie.image, genre: movie.genre, times_viewed: movie.times_viewed, comments: movie.comments}),
+        enableReinitialize: true,
+        validationSchema: validationSchema,
+        onSubmit: values => {
+            dispatch(editMovie(values));
+            handleClose();
+        },
+    });
+  
+    return (
+      <>
+        <button className="float-right m-1" onClick={handleShow}>
+          <i className='fas fa-edit'></i>Edit Movie
+        </button>
+  
+        <Modal show={show} onHide={handleClose}>
+          <Modal.Header closeButton>
+            <Modal.Title>Edit Movie</Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Loader isLoading={!movie}>{ () =>
+          <Form>
+            <Form.Row className="text-left">
+                <Form.Group as={Col}>
+                <Form.Label>Title</Form.Label>
+                <Form.Control 
+                    type="text" 
+                    required
+                    id="title"
+                    {...formik.getFieldProps('title')}
+                    />
+                {formik.touched.title && formik.errors.title ? (
+                        <div style={{'color':'red'}}>*{formik.errors.title}</div>
+                    ) : null}
+                </Form.Group>
+            </Form.Row>
+            <Form.Row>
```

Comment:
Jel cela forma ista kao i create forma? Da li su mogli da dele to? Mozda bude komponenta pa editMovieModal i CreateMovieModal da je dele, da je samo wrapuju ali mi nesto deluej da vecina koda moze da se deli? Neka komponenta koji je samo prezentaciona mozda?